masonry/src/widget/variable_label.rs

```diff
@@ -0,0 +1,393 @@
+// Copyright 2019 the Xilem Authors and the Druid Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//! A label with support for animated variable font properties
+
+use accesskit::Role;
+use kurbo::{Affine, Point, Size};
+use parley::fontique::Weight;
+use parley::layout::Alignment;
+use parley::style::{FontFamily, FontStack};
+use smallvec::SmallVec;
+use tracing::{trace, trace_span, Span};
+use vello::peniko::BlendMode;
+use vello::Scene;
+
+use crate::text::{TextBrush, TextLayout, TextStorage};
+use crate::widget::WidgetMut;
+use crate::{
+    AccessCtx, AccessEvent, ArcStr, BoxConstraints, EventCtx, LayoutCtx, LifeCycle, LifeCycleCtx,
+    PaintCtx, PointerEvent, StatusChange, TextEvent, Widget, WidgetId,
+};
+
+use super::LineBreaking;
+
+// added padding between the edges of the widget and the text.
+pub(super) const LABEL_X_PADDING: f64 = 2.0;
+
+/// An `f32` value which can move towards a target value at a linear rate over time.
+#[derive(Clone, Debug)]
+pub struct AnimatedF32 {
+    /// The value which self will eventually reach.
+    target: f32,
+    /// The current value
+    value: f32,
+    // TODO: Provide different easing functions, instead of just linear
```

Comment:
Maybe a future case for interpoli?

I think this could also be achieved with the `Animatable` trait I did in trui (to be more generally customizable in xilem and I think also in masonry, but I haven't looked into that yet... (also because I have not yet gotten real feedback there)).

Reply:
Yeah, this would be the type of thing I expect Interpoli to provide.
I haven't had time to dig into that Animatable work, unfortunately.